File: Strivers/src/basicsmaths/lcmandgcd/LcmAndGcd.java
package basicsmaths.lcmandgcd;

import java.util.Arrays;
import java.util.Scanner;

public class LcmAndGcd {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        long a = sc.nextLong();
        long b = sc.nextLong();
        System.out.println(Arrays.toString(lcmAndGcd(a, b)));
    }

    static Long[] lcmAndGcd(Long A , Long B) {
        // code here
        long gcd = gcd(A, B);
        long lcm = A * B / gcd;
        return new Long[] {lcm, gcd};
    }

    static long gcd(long a, long b) {
        while(a != b) {
            if(a > b) a -= b;
            else b -= a;
        }
        return a;
    }
}

File: Strivers/src/basicsmaths/lcmandgcd/LcmAndGcd.kt
package basicsmaths.lcmandgcd

import java.util.Scanner

fun main(args: Array<String>) {
    val sc = Scanner(System.`in`)
    val a: Long = sc.nextLong()
    val b: Long = sc.nextLong()
    println(lcmAndGcd(a ,b).contentToString())
}

fun lcmAndGcd(a: Long, b: Long): Array<Long> {
    val gcd = gcd(a, b)
    val lcm = a * b / gcd
    return arrayOf(lcm, gcd)
}

fun gcd(a: Long, b: Long): Long {
    var a1 = a
    var b1 = b
    while(a1 != b1) {
        if (a1 > b1) a1 -= b1 else b1 -= a1
    }
    return a1
}